File: frontend/src/services/categoryService.js
import api from "./api";
 
export const addCategory = async (categoryData) => {
  const response = await api.post("/api/categories/addCategory", categoryData);
  return response.data;
};
 
export const updateCategory = async (id, categoryData) => {
  const response = await api.put(`/api/categories/updateCategory/${id}`, categoryData);
  return response.data;
};
 
export const deleteCategory = async (id) => {
  await api.delete(`/api/categories/deleteCategory/${id}`);
};
 
export const getCategoryById = async (id) => {
  const response = await api.get(`/api/categories/getCategoryById/${id}`);
  return response.data;
};
 
export const getAllCategories = async () => {
  const response = await api.get("/api/categories/getAllCategories");
  return response.data;
};
 
export const getCustomCategoryByUser = async (userId) => {
  const response = await api.get(`/api/categories/getCustomCategoryByUser/${userId}`);
  return response.data;
};

File: backend/Expense_tracker_service/java/com/monexel/expensetracker/service/CategoryService.java
package com.monexel.expensetracker.service;

import java.util.List;

import com.monexel.expensetracker.request.CategoryRequest;
import com.monexel.expensetracker.response.CategoryResponse;

public interface CategoryService {

	CategoryResponse addCategory(CategoryRequest request);

	CategoryResponse updateCategory(Long id, CategoryRequest request);

	void deleteCategory(Long id);

	CategoryResponse getCategoryById(Long id);

	List<CategoryResponse> getAllCategories(); // Predefined + custom

	List<CategoryResponse> getCustomCategoriesByUser(Long userId);

}
